ui/src/pages/backoffice/Patron/PatronDetails/components/ItemsSearch/ItemsSearch.js

```diff
@@ -81,111 +64,79 @@ export default class ItemsSearch extends Component {
     }
   };
 
-  renderSearchBar = () => {
-    return (
-      <Input
-        action={{
-          content: 'Search',
-          onClick: () => {
-            this.executeSearch();
-          },
-        }}
-        fluid
-        focus
-        placeholder={'Search by barcode...'}
-        onChange={(e, { value }) => {
-          this.onInputChange(value);
-        }}
-        onPaste={e => {
-          this.onPasteHandler(e);
-        }}
-        value={this.props.queryString}
-        onKeyPress={event => {
-          this.onKeyPressHandler(event.target);
-        }}
-        ref={input => {
-          this.searchInput = input;
-        }}
-      />
-    );
-  };
-
   renderResultsList = results => {
     return (
       <div className="results-list">
-        {this.renderHeader(results.hits.length)}
         <ItemsResultsList
           patronPid={this.props.patronDetails.user_pid}
           clearResults={this.clearResults}
           results={results}
-          checkoutItem={this.checkoutItem}
-          fetchPatronCurrentLoans={this.fetchUpdatedCurrentLoans}
+          clearSearchQuery={this.clearSearchQuery}
         />
       </div>
     );
   };
 
-  renderEmptyResults = (queryString, resetQuery) => {
+  clearSearchQuery = () => {
+    this.setState({ prevSearchQuery: '' });
+    this.clearResults();
+  };
+
+  renderSearchPrompt = () => {
     return (
       <Segment placeholder textAlign="center">
         <Header icon>
           <Icon name="search" />
-          Found no items matching this barcode.
+          No barcode provided.
         </Header>
         <div className="empty-results-current">
-          Current search phrase "{queryString}"
+          Type or paste the barcode to search for items
         </div>
-        <Segment.Inline>
-          <Button primary onClick={() => resetQuery('')}>
-            Clear query
-          </Button>
-        </Segment.Inline>
       </Segment>
     );
   };
 
-  renderHeader = totalResults => {
-    return <p>Found {totalResults} item(s).</p>;
-  };
-
   render() {
-    const { items, isLoading, error } = this.props;
-    const { prevSearchQuery } = this.state;
+    const { items, isLoading, error, queryString } = this.props;
     return (
-      <Segment className={'patron-items'}>
-        <Header as={'h3'}>Items</Header>
-        <Header.Subheader>Search items by barcode.</Header.Subheader>
-        <Container className={'search-bar'}>{this.renderSearchBar()}</Container>
+      <>
+        <Container className="search-bar spaced">
+          <SearchBar
+            action={{
+              icon: 'search',
+              onClick: () => this.executeSearch(),
```

Comment:
can we do `onClick: this.executeSearch`?

Reply:
unfortunately no, because I want it to be called without params. if I do it your way it will pass the event as a param. Unless you know a better way?

Reply:
Good, then, can you please add a comment or wrap it in another function?
```
onClick: this.executeSearchWithoutParams()
```
I feat that we will forget why and we will refactor it at some point.